### PR TITLE
feat(recommendation): make recommendation have publishDate property

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding a `published` property on extension recommended, to only display them past the date.

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-internal/issues/242

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

File: packages/main/src/plugin/recommendations/recommendations-api.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export interface ExtensionBanner {
   description: string;
   icon: string;
   thumbnail: string;
+  published?: string;
```
Comment (deboer-tim) [inline packages/main/src/plugin/recommendations/recommendations-api.ts]: With just the word `published` I expect this to be a boolean, like 'is it published or not?' I would need to find docs or an example to understand what to put here. Maybe should be `publishDate` or `embargoUntil` to more clearly state its purpose?

**Review by deboer-tim — COMMENTED** [review]
Code and function LGTM, but I think we should rename.

File: packages/main/src/plugin/recommendations/recommendations-api.ts (hunk 1 above)
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-api.ts]:
```suggestion
  publishDate?: string;
```

File: packages/main/src/plugin/recommendations/recommendations-registry.spec.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ vi.mock('../../../../../recommendations.json', () => ({
       description: 'dummy description',
       icon: 'data:image/png;base64-icon',
       thumbnail: 'data:image/png;base64-thumbnail',
+      published: '2020-01-01',
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.spec.ts]:
```suggestion
      publishDate: '2020-01-01',
```

File: packages/main/src/plugin/recommendations/recommendations-registry.spec.ts [hunk 3]
```diff
@@ -190,4 +201,26 @@ describe('getExtensionBanners', () => {
 
     expect(featuredMock.getFeaturedExtensions).toHaveBeenCalled();
   });
+
+  test('published value anterior', async () => {
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.spec.ts]:
```suggestion
  test('publishDate value anterior', async () => {
```

File: packages/main/src/plugin/recommendations/recommendations-registry.ts [hunk 4]
```diff
@@ -50,15 +50,28 @@ export class RecommendationsRegistry {
...
-      }))
-      .sort(() => Math.random() - 0.5);
+      .reduce((prev, extension) => {
+        // ensure the extension is in the featured extensions and is not install
+        if (!(extension.extensionId in featuredExtensions) || featuredExtensions[extension.extensionId].installed) {
+          return prev;
+        }
+
+        // Check for published property
+        if ('published' in extension && typeof extension.published === 'string') {
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.ts]:
```suggestion
        if ('publishDate' in extension && typeof extension.publishDate === 'string') {
```

File: packages/main/src/plugin/recommendations/recommendations-registry.ts [hunk 5]
```diff
@@ -50,15 +50,28 @@ export class RecommendationsRegistry {
...
+          const published = new Date(extension.published).getTime();
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.ts]:
```suggestion
          const publishDate = new Date(extension.publishDate).getTime();
```

File: packages/main/src/plugin/recommendations/recommendations-registry.ts [hunk 6]
```diff
@@ -50,15 +50,28 @@ export class RecommendationsRegistry {
...
+          if (isNaN(published) || published > Date.now()) {
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.ts]:
```suggestion
          if (isNaN(publishDate) || publishDate > Date.now()) {
```

File: packages/main/src/plugin/recommendations/recommendations-registry.ts [hunk 7]
```diff
@@ -50,15 +50,28 @@ export class RecommendationsRegistry {
...
-        featured: featuredExtensions[extension.extensionId],
-      }))
-      .sort(() => Math.random() - 0.5);
+      .reduce((prev, extension) => {
+        // ensure the extension is in the featured extensions and is not install
+        if (!(extension.extensionId in featuredExtensions) || featuredExtensions[extension.extensionId].installed) {
+          return prev;
+        }
+
+        // Check for published property
```
Comment (axel7083) [inline packages/main/src/plugin/recommendations/recommendations-registry.ts]:
```suggestion
        // Check for publishDate property
```

File: packages/main/src/plugin/recommendations/recommendations-api.ts (hunk 1 above)
Comment (axel7083) [reply packages/main/src/plugin/recommendations/recommendations-api.ts]: Yes I agree, done ✅

**Review by deboer-tim — APPROVED** [review]
Thanks.

**Pull request merged**